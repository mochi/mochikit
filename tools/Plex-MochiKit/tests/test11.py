import Test
from Plex import *
import sys

lex = Lexicon([
  (Str("Python"), 'upper-python'),
  (Str("python"), 'lower-python'),
  (NoCase(Str("COBOL", "perl", "Serbo-Croatian")), 'other-language'),
  (NoCase(Str("real") + Case(Str("basic"))), 'real-1'),
  (NoCase(Str("real") + Case(Str("Basic"))), 'real-2'),
  (Any(" \t\n"), IGNORE)
],
debug = Test.debug,
timings = sys.stderr
)

Test.run(lex, "test11", debug = 0, trace = 0)
