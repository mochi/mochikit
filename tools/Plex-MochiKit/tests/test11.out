test11.in   1   0 upper-python 'Python'
test11.in   1   7 lower-python 'python'
test11.in   2   0 other-language 'COBOL'
test11.in   2   6 other-language 'cobol'
test11.in   2  12 other-language 'CoBol'
test11.in   2  18 other-language 'COboL'
test11.in   3   0 other-language 'perl'
test11.in   3   5 other-language 'Perl'
test11.in   3  10 other-language 'pERl'
test11.in   4   0 other-language 'Serbo-Croatian'
test11.in   4  15 other-language 'serbo-croatian'
test11.in   5   0 real-1     'REALbasic'
test11.in   5  10 real-1     'realbasic'
test11.in   5  20 real-1     'REalbasic'
test11.in   6   0 real-2     'REALBasic'
test11.in   6  10 real-2     'realBasic'
test11.in   6  20 real-2     'REalBasic'
test11.in   7   0 None       ''
